### ISSUE TITLE
DOC: Minimum Python Version?

### ISSUE THREAD

**Issue by chrisyeh96** [zach401/acnportal#106]
Is there any minimum Python version required for ACN portal? Clearly, Python 3.6 is necessary for using f-strings. But ideally, if you target Python 3.7+, then you can simplify a lot of the typing annotations. For example, you can do things like the following in Python 3.7+:

```python
from __future__ import annotations

def f(x: dict[str, int] | None = None) -> dict | None:
    return x
```

I didn't see any minimum Python version specified in the docs, so if that can be added (and ideally Python 3.7+), then I could help add appropriate typing annotations.

**Comment by sunash** [comment]
Hi @chrisyeh96, I'm okay with a 3.7 minimum version. I think it would probably need to be a new release, though. @zach401 what do you think? 

I'm personally not familiar with __future__.annotations; are there any major functional differences between that and the typing package? Namely, I assume mypy allows either?

**Comment by chrisyeh96** [comment]
The [`__future__` package](https://docs.python.org/3/library/__future__.html) in the Python standard library contains experimental ideas, essentially for beta testing. It enables the following 3 main changes:
* you can directly use built-in Python types like `list` and `dict` in your typing annotations, without having to call `from typing import List, Dict`
* you can use `str | None` instead of writing `from typing import Optional`, then `Optional[str]`
* you can use `str | dict` instead of writing `from typing import Union`, then `Union[str, dict]`

In short, it just makes the typing annotations look cleaner. The benefits are purely cosmetic.

(There are actually a few other nontrivial situations where `from __future__ import annotations` actually helps with, but they are rare, and I won't explain them here.)
